lua/ttt2/libraries/game_effects.lua

```diff
@@ -0,0 +1,313 @@
+---
+-- A library to consolidate some common effects code.
+-- @author EntranceJew
+-- @module game_effects
+if SERVER then
+	AddCSLuaFile()
+end
+
+game_effects = game_effects or {}
+game_effects.extinguishables = {"ttt_flame", "vfire", "vfire_cluster", "vfire_ball", "entityflame", "env_fire"}
+
+---
+-- @param Entity ent The entity whomst's anti-flammability is under question.
+-- @realm shared
+function game_effects.IsExtinguishableEntity(ent)
+	return (ent.GetClass and table.HasValue(game_effects.extinguishables, ent:GetClass()))
+		or ent:IsOnFire()
+end
+
+---
+-- @param Entity ent The entity to remove flames from.
+-- @realm shared
+function game_effects.Extinguish(ent)
+	local class = ent:GetClass()
+	---
+	-- provided by rubat's extinguisher
+	-- @realm shared
+	local retval = hook.Run("ExtinguisherDoExtinguish", ent)
```

Comment:
Is this a new hook? or what is this?

Reply:
Rubat's extinguisher does this, and it's still one of the most popular "extinguishers", so subsequently a lot of things that can be extinguished by other means have hooked on to this. i included it for the sake of compatibility, but i'm not sure if it needs anything else documenting its existence.

Reply:
ah, then please add docs ala: `function GM:ExtinguisherDoExtinguish(ent)` with doc strings above with a link to the source. So it is clear and visible in the docs on: https://api-docs.ttt2.neoxult.de/